Cleanup nRF53 cpunet enabling code

gmarull opened the pull request:
Initial cleanup on nRF53 cpunet enabling code. It looks like a lot of duplication was present, making things unnecessarily complicated. This is a first round of cleanups, things can likely be improved further.

gmarull committed:
bluetooth: controller: nordic: simplify board guarding

Since HWMv1, we also have CONFIG_$BOARD Kconfig symbols defined, ie, no
SoC/core/variant needed.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
bluetooth: controller: nordic: provide dummy DEBUG_SETUP

So that clients do not have to understand in which situation it is not
defined.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
drivers: bluetooth: hci: nrf53: simplify DEBUG_SETUP code

debug.h module provides dummy implementation when necessary.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
soc: nordic: nrf53: add option to enable cpunet

When not using BT, users may want to enable the cpunet core. Until now,
this has been done at board level (so duplicating unnecessary code)
using CONFIG_BOARD_ENABLE_CPUNET. The board-level options were usually
enabled automatically for BT, however, this was unnecessary as BT driver
already takes care of the setup.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
drivers: bluetooth: hci: nrf53: use nrf_spu_extdomain_set

Instead of hardcoded values and device instances.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
boards: nrf53: remove redundant cpunet initialization code

This is now handled at SoC level, avoiding code duplicates.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
soc: nordic: nrf53: introduce deprecated BOARD_ENABLE_CPUNET

So that existing code does not break when this option gets removed from
boards.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
drivers: clock_control: nrf: use SOC_NRF53_CPUNET_ENABLE

Board level option is deprecated.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
samples: s/CONFIG_BOARD_ENABLE_CPUNET/CONFIG_SOC_NRF53_CPUNET_ENABLE/

CONFIG_BOARD_ENABLE_CPUNET is deprecated.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

cvinayak commented on drivers/bluetooth/hci/nrf53_support.c:
```diff
@@ -30,16 +29,15 @@ int bt_hci_transport_teardown(const struct device *dev)
 int bt_hci_transport_setup(const struct device *dev)
 {
 	ARG_UNUSED(dev);
-#if !defined(CONFIG_TRUSTED_EXECUTION_NONSECURE) || defined(CONFIG_BUILD_WITH_TFM)
```
GPIO setup may fail? (but will go unnoticed, `DEBUG_SETUP` may not be covered in CI.

gmarull replied:
`DEBUG_SETUP()` is now a macro expanding to a set of unchecked calls, so as it is designed, can't fail.

cvinayak commented on subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/debug.h:
What have you done here? CPUAPP needs to forward, and CPUNET needs to define the GPIOs.

gmarull replied:
Github is not pointing to any change here hmm... what do you mean?

cvinayak replied:
my misunderstanding.... i thought you combined CPUAPP and CPUNET cases together... but I am in sync now.

masz-nordic commented on soc/nordic/nrf53/nrf53_cpunet_mgmt.c:
```diff
@@ -66,3 +68,33 @@ void nrf53_cpunet_enable(bool on)
 }
+
+#ifdef CONFIG_SOC_NRF53_CPUNET_ENABLE
+static int nrf53_cpunet_init(void)
```
Looks quite similar to what we are doing with VPRs on 54s.
A common way to initialize and run secondary cores in Zephyr would be useful.

gmarull replied:
agreed, a remoteproc-like API should be introduced. I'll look into this post-release.

anangl commented on boards/nordic/nrf5340dk/CMakeLists.txt:
```diff
@@ -2,14 +2,8 @@
+    AND CONFIG_BUILD_WITH_TFM)
   zephyr_library_include_directories(
     $<TARGET_PROPERTY:tfm,TFM_BINARY_DIR>/api_ns/interface/include
   )
```
It should be safe to remove this as well. If I understand 88bb3ae498426d83b560acdd7470fd378e693d90 correctly, this was only introduced because at that time `soc_secure.h` [was including `tfm_ioctl_api.h`](https://github.com/zephyrproject-rtos/zephyr/blob/88bb3ae498426d83b560acdd7470fd378e693d90/soc/arm/nordic_nrf/nrf53/soc_secure.h#L12) , so that path was missing for `subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/debug.h` that was included by `nrf5340_cpunet_reset.c`. But after soc_secure refactoring in 6a14a0233b7eb6d66d4380992d2e292470e05a72, this became unneeded.
Same applies to `bl5340_dvk` and `nrf5340_audio_dk`.

gmarull replied:
added one more commit to delete these includes

gmarull committed:
boards: nrf53*: remove unnecessary TF-M includes

After 6a14a0233b7eb6d66d4380992d2e292470e05a72 these TF-M includes
should not be necessary.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

nordic-krch commented on soc/nordic/nrf53/nrf53_cpunet_mgmt.c:
```diff
@@ -66,3 +68,33 @@ void nrf53_cpunet_enable(bool on)
+static int nrf53_cpunet_init(void)
+{
+	/* Route Bluetooth Controller Debug Pins */
+	DEBUG_SETUP();
```
do you need that here? `CONFIG_SOC_NRF53_CPUNET_ENABLE` depends on `!BT` so it seems strange that bluetooth debug pins are set up here.

gmarull replied:
hmm you're right, good catch

dchat-nordic commented:
Will this PR be merged soon?

jhedberg commented:
> Will this PR be merged soon?

It's missing approval by the assignee (@anangl)
